storage/statedb/database.go

```diff
@@ -1129,28 +1128,49 @@ func (db *Database) UpdateMetricNodes() {
 var errDisabledTrieNodeCache = errors.New("trie node cache is disabled, nothing to save to file")
 var errSavingTrieNodeCacheInProgress = errors.New("saving trie node cache has been triggered already")
 
-// SaveTrieNodeCacheToFile saves the current cached trie nodes to file to reuse when the node restarts
-func (db *Database) SaveTrieNodeCacheToFile(filePath string) error {
+func (db *Database) CanSaveTrieNodeCacheToFile() error {
 	if db.trieNodeCache == nil {
 		return errDisabledTrieNodeCache
 	}
 	if db.savingTrieNodeCacheTriggered {
 		return errSavingTrieNodeCacheInProgress
 	}
+	return nil
+}
+
+// SaveTrieNodeCacheToFile saves the current cached trie nodes to file to reuse when the node restarts
+func (db *Database) SaveTrieNodeCacheToFile(filePath string, concurrency int) {
 	db.savingTrieNodeCacheTriggered = true
 	start := time.Now()
-	go func() {
-		logger.Info("start saving cache to file", "filePath", filePath)
-		if err := db.trieNodeCache.SaveToFile(filePath, runtime.NumCPU()/2); err != nil {
-			logger.Error("failed to save cache to file",
-				"filePath", filePath, "elapsed", time.Since(start), "err", err)
-		} else {
-			logger.Info("successfully saved cache to file",
-				"filePath", filePath, "elapsed", time.Since(start))
+	logger.Info("start saving cache to file",
+		"filePath", filePath, "concurrency", concurrency)
+	if err := db.trieNodeCache.SaveToFile(filePath, concurrency); err != nil {
+		logger.Error("failed to save cache to file",
+			"filePath", filePath, "elapsed", time.Since(start), "err", err)
+	} else {
+		logger.Info("successfully saved cache to file",
+			"filePath", filePath, "elapsed", time.Since(start))
+	}
+	db.savingTrieNodeCacheTriggered = false
+}
+
+// DumpPeriodically atomically saves fast cache data to the given dir with the specified interval.
+func (db *Database) SaveCachePeriodically(c *TrieNodeCacheConfig, stopCh <-chan struct{}) {
+	ticker := time.NewTicker(c.FastCacheSavePeriod)
```

Comment:
How about setting random value at the 1st only?

```suggestion
	ticker := time.NewTicker(c.FastCacheSavePeriod)
```

Reply:
@KimKyungup If you're worried about consistent performance degradation after the reboot due to round change, I have no objection. How long do you think it is proper for random period? (1hour? FastCacheSavePeriod?)

Reply:
Reflected by f9e2dd790d6589e8e10b7d690dbf634dee0cc774, please check @KimKyungup

Reply:
@ehnuje Thanks. :)